crates/circuit/src/dag_node.rs

```diff
@@ -110,34 +119,16 @@ impl DAGOpNode {
             }
             None => qargs.str()?.into_any(),
         };
-        let res = convert_py_to_operation_type(py, op.clone_ref(py))?;
 
-        let extra_attrs = if res.label.is_some()
-            || res.duration.is_some()
-            || res.unit.is_some()
-            || res.condition.is_some()
-        {
-            Some(Box::new(ExtraInstructionAttributes {
-                label: res.label,
-                duration: res.duration,
-                unit: res.unit,
-                condition: res.condition,
-            }))
-        } else {
-            None
-        };
+        let mut instruction = CircuitInstruction::py_new(
+            py, op, None, None, params, label, duration, unit, condition,
+        )?;
+        instruction.qubits = qargs.into();
+        instruction.clbits = cargs.into();
```

Comment:
It's because of a type mismatch. The constructor for a circuit instruction takes in an `impl IntoIterator` while we have a `PyTuple` here which isn't doesn't implement the `IntoItertator` trait. I thought about collecting the tuples into a `Vec` so we could pass them to `new()`, but at the end of the day the `CircuitInstruction` has a `PyTuple` in it, so I just elected to do this to avoid needing any conversions or tweaks to the interfaces.